locations/locations.go

```diff
@@ -37,12 +39,24 @@ func pathLocation(d desc.Descriptor, path ...int) *dpb.SourceCodeInfo_Location {
 	return sourceInfoRegistry.sourceInfo(d.GetFile()).findLocation(fullPath)
 }
 
-type sourceInfo map[string]*dpb.SourceCodeInfo_Location
+type sourceInfo struct {
+	info map[string]*dpb.SourceCodeInfo_Location
+	lock sync.Mutex
```

Comment:
The "mutex hat" pattern prefers the mutex above, and with a name that indicates what it controls: https://github.com/go-critic/go-critic/issues/465

Reply:
Nice! Done.